### PR TITLE
Fix silent failure/exception with aws s3 sync script and other script improvements

### PR DESCRIPTION
**Pull request by kaipak**
## 🗒️ Summary
PR fixes silent aws s3 sync issue with script along with a few other minor issues related to #33. Original code replaced with try/exception so that error messages are caught and outputted to stdout. This will be sent to logging subsystem with later PR.

- Fix silent failures with try/exception structure
- Add code comments and comment blocks to every class/method
- Add parse args feature so script can accept command line arguments now. Hard-coded config file reference removed.
- Other minor improvements related to code readability and maintenance.

## ⚙️ Test Data and/or Report
NA. Testing framework is in a separate effort and will be created there. Please see #35 

## ♻️ Related Issues
- #36 
- #35 
- #33

### CODE REVIEW

File: config/config_dev_kai.yaml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    geo-va:
+      include:
+        - "u_ex24*"
+  img:
+    img-pdsimage-http:
+      include:
+        - "*.2024*"
+    img-pdsimg1-http:
+      include:
+        - "*.2024*"
```
Comment (c-suh) [inline config/config_dev_kai.yaml:42]: With their new machines, this log filename has changed such that this pattern doesn't work anymore. It should be changed to "*-2024*".

File: config/config_dev_kai.yaml [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    img-pdsimg2-http:
+      include:
+        - "*.2024*"
```
Comment (c-suh) [inline config/config_dev_kai.yaml:45]: See comment 3 lines above.

File: config/config_dev_kai.yaml [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    img-pdsimg3-http:
```
Comment (c-suh) [inline config/config_dev_kai.yaml:46]: With their new machines, this profile has been removed.

**Review by c-suh — COMMENTED** [review]
Not specific to the task, but I noticed that the regex for some of the img profiles is incorrect. I'm also wondering if some profiles (e.g., sbn tools, atm ftp, geo ftp) were purposely left out? If this wasn't on purpose, we can talk specifics via Slack.

Also, what is the difference between this PR and [PR#36](https://github.com/NASA-PDS/web-analytics/pull/36)?

**Comment by kaipak** [comment]
Sorry, those config files shouldn't really be checked in so I've removed from cached history. Review should be for the code changes mainly. I closed #36 just as a formality because I was trying to merge from a feature branch instead of dev.

**Review by c-suh — APPROVED** [review]
@kaipak tested on my account on the new machine. It initially gave an ExpiredToken error, and after that was addressed, showed the files syncing! This looks to be working as intended.

**Pull request merged**